### ISSUE TITLE
NaN gradient after py_func call

### ISSUE THREAD

**Issue by Heungwoo** [tensorflow/tensorflow#5250]
Hello,

I did implemented faster rcnn with py_func op. 
- py-func is placed in the middle of graph. And there are the other path in parallel with py_func node.
- I did not add any gradient function for it, So it should ignore gradient path through py_func. And gradient calculation should done with connection from the other path.

It works fine in tensorflow r0.9 version. 
But, in the tensorflow version > r0.9, the node before py_func get NaN error when writing histogram summary. .
I did check several version: r0.10, r0.11, all of them does not worked.

Is there any change in gradient calculation for the node which have connection with py_func node ?
Or should I use py-func in different way?

Any advice will be helpful for me.
Thanks in advance.

**Comment by drpngx** [comment]
Do you happen to have a small repro case?

**Comment by Heungwoo** [comment]
It is not the pyfunc which makes NaN value.
Softmax_with_cross_entropy funcution was changed in version 0.10 and it return NaN if label gt is not in valid label value range. I did you "-1" label as ignore-label in version 0.9 and just use valid index value to calculate loss. But migrating it to upper version, it makes NaN loss.

Sorry to make mistake.

**Comment by drpngx** [comment]
Good to know. Glad you figured it out!
